packages/fxa-auth-server/lib/routes/account.ts

```diff
@@ -1074,11 +1074,49 @@ export class AccountHandler {
       return false;
     };
 
-    const skipTokenVerification = (request: AuthRequest, account: any) => {
+    const skipTokenVerification = async (request: AuthRequest, account: any) => {
       // Skip all checks to simulate an unverified session token state
       if (this.config.signinConfirmation.tokenVerification === false) {
         return false;
       }
+
+      // Check to see if there has been a recent, successfully-verified login from
+      // the same device (as indicated by User-Agent string)
+      if (this.config.signinConfirmation.deviceFingerprinting?.enabled) {
+        try {
+          const currentUserAgent = request.headers['user-agent'] || '';
+          const skipTimeframeMs = this.config.signinConfirmation.deviceFingerprinting.duration as unknown as number;
+
+          const isRecognized = await isRecognizedDevice(this.db, account.uid, currentUserAgent, skipTimeframeMs);
```

Comment:
Should we add an exclusion if the device was explicitly signed out since the last successful verification (for example for (stolen) devices disconnected from the Connected services list)?

Reply:
Also wondering if the login confirmation email should include a link to disconnect the device if it is unrecognized 🤔

Reply:
I would probably hold off on doing an exclusion list since we currently don't do that if an ip address was signed out. Seems reasonable to consider it in a follow up.

The user would still get the "New Device Login" email and that has a link to settings to manage devices. Maybe that is ok?